File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        
 
        
        int size = nums.size();
        int b = k%size;
        vector<int>ans(size);
        
        for(int i =0; i<size;i++){
            
            ans[b] = nums[i];
            b = (b+1)%size;
        }
        
        nums = ans;


        
    }
};

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        k = k% len(nums)
        nums[:] = nums[-(k+1)::-1] + nums[-1:-(k+1):-1]
        nums[:] = nums[::-1]
        